### DISCUSSION TITLE
SFTP Working Directory

### DISCUSSION THREAD

**Discussion by JacobPellegrino** [Eugeny/tabby#5558] [Q&A]
Is there a way to make the SFTP window default to working directory?

**Comment by Eugeny** [comment]
Click the lightbulb icon in the toolbar - it will give you directions on how to set up working directory detection.

**Reply by JacobPellegrino** [reply]
Thank you! Where do we access the shell profile file? I just tried the profiles and connections tab and am not sure which option there or if it's an actual system file.

**Reply by Eugeny** [reply]
On the remote server - use `vi` or another editor to edit the file.

**Reply by JacobPellegrino** [reply]
I'm sorry for all the questions but what is the file called / which directory would I find it in (generally). Thanks again.

**Reply by Eugeny** [reply]
No worries. It's `~/.bash_profile` for bash and `~/.zshrc` for zsh - those are full paths - the files are in your home directory.

**Reply by HelloRicky123** [reply]
Thanks for the solution. But this is not effective when using tmux. Do you have suggestions for reporting the working directory in tmux?

**Comment by thudjx** [comment]
Hey, I found that working directory works perfectly after add the command line as instructed, but it is not compatible with tmux. In fact, it reports the directory when the `tmux` command is invoked, i.e., tmux session is called. Is there any way to make the working directory reporting compatible with tmux?

**Reply by ywgrit** [reply]
> Hey, I found that working directory works perfectly after add the command line as instructed, but it is not compatible with tmux. In fact, it reports the directory when the `tmux` command is invoked, i.e., tmux session is called. Is there any way to make the working directory reporting compatible with tmux?

Hi, Did you solve the problem?

**Comment by sourav026** [comment]
I have apllied the changes mentioned in the below link,
https://github.com/Eugeny/tabby/wiki/Shell-working-directory-reporting

but it still not working for me. And FYI i did not have .zshrc file and config.fish file. I had to create both those files manually.

**Reply by han-minhee** [reply]
It's been more than a year, so I guess you already found a solution or so. The modification should be done according to which shell environment you're using. The absence of zshrc or fish file means that you're not using them. My guess is that your shell environment wasn't bash/zsh or fish.
